utils/print_queries.f90

```diff
@@ -58,18 +57,19 @@ subroutine print_queries(input_file, subset)
       do subset_idx = 1, size(subsets)
         print *, "  ", subsets(subset_idx)%chars()
       end do
+      print *, "Total number of subsets found:", size(subsets)
       print *, ""
 
+      ! Reset the file
+      call closbf(FileUnit)
+      close(FileUnit)
+      open(FileUnit, file=trim(input_file))
+      call openbf(FileUnit, "IN", FileUnit)
```

Comment:
I would undo this change and move this block back inside the `do` loop. The reason for this is that 1 message can have several subsets, and our BUFR files may be concatenated so there are very few messages in it. If you don't reset the file pointer between subsets you run the risk of running out of messages before you find an example for every subset table to print (safest to act like there may only be 1 message in the BUFR file).

Reply:
@rmclaren 

Would adding the "rewind" in the do-loop work too?
This can avoid closing/opening file repeatedly.  

```
     ! Reset the file
      call closbf(FileUnit)
      close(FileUnit)
      open(FileUnit, file=trim(input_file))
      call openbf(FileUnit, "IN", FileUnit)

      do subset_idx = 1, size(subsets)

        rewind(FileUnit)

        q_paths = all_queries(FileUnit, subsets(subset_idx))

        print *, "Possible queries for subset: ", subsets(subset_idx)%chars()
        call print_paths(q_paths)
        print *, ""

      end do
```